connectors/sync_job_runner.py

```diff
@@ -516,19 +522,46 @@ async def generator(self):
                 raise UnsupportedJobType
 
     async def update_ingestion_stats(self, interval):
+        last_cursor = None
         while True:
             await asyncio.sleep(interval)
 
             if not await self.reload_sync_job():
                 break
 
+            cursor = (
+                None
+                if not self.data_provider  # If we failed before initializing the data provider, we don't need to change the checkpoint
+                else (
+                    self.data_provider.sync_cursor()
+                    if self.sync_job.is_content_sync()
+                    else None
+                )
+            )
+
+            if cursor != last_cursor:
```

Comment:
```suggestion
            if cursor != last_cursor and cursor != None:
```

I don't think we want to overwrite an old cursor if this was an access-control sync or if the data_provider failed to initialize